doc/contributor/workarounds.md

```diff
@@ -0,0 +1,37 @@
+# Patching system
+
+The patching system is loaded in `post-boot.rb` and is able to patch stdlib and installed gems. 
+The code implementing the patching itself can be found in `lib/truffle/truffle/patching.rb`.
+It can be disabled by passing the `-Xpatching=false` option.
+
+The patching system works for gems as follows. 
+When a gem *g* is activated, 
+there is a directory `g` in `lib/patches`, 
+and the directory is listed in `lib/truffle/truffle/patching.rb`,  
+the directory is injected into `$LOAD_PATH` before the original load-paths of the *g* gem. 
```

Comment:
Minor, `inserted` here might be better than `injected`, implicitly referring to Array#insert.